[ADD] crm: expected revenue report doc (#7980)
odoo/documentation: content/applications/sales/crm/performance/expected_revenue_report.rst

=======================
Expected revenue report
=======================

> Expected Revenue is the total cash value of leads that are expected to close by a certain date,

Reviewer: ```suggestion
*Expected Revenue* is the total cash value of leads that are expected to close by a certain date,
```

usually the end of the current month.

An *expected revenue report* compiles all active leads in a sales pipeline that have a set expected
closing date, and compares how sales teams are performing in a given month.

.. image:: expected_revenue_report/expected-revenue-closing.png
   :align: center
   :alt: Close up of the expected closing date on a lead in the CRM app.

By pulling a monthly expected revenue report, sales managers can see which team members are reaching

> goals and which may need additional assistance to close valuable deals.

Reviewer: ```suggestion
goals, and which may need additional assistance to close valuable deals.
```

Create an expected revenue report
=================================

To create an expected revenue report, first navigate to :menuselection:`CRM app --> Reporting -->
Pipeline`. This opens the :guilabel:`Pipeline Analysis` report.

.. important::
   The *Pipeline Analysis* report includes several filters in the search bar by default. Remove
   these before adding any additional custom filters.

On the top-left of the report, click :guilabel:`Measures`, then select :guilabel:`Expected Revenue`
from the drop-down menu.

> At the top of the page, click the arrow to the right of the search bar to open the

Reviewer: ```suggestion
At the top of the page, click the :guilabel:`down arrow` icon to the right of the search bar to open the
```

:guilabel:`Filters` drop-down menu. Select :guilabel:`Add a Custom Filter`.

Add custom filters
------------------

On the :guilabel:`Add Custom Filter` pop-up window, click into the first field of the new rule.
Type `Expected Closing` into the search bar, or scroll to select it from the list. Click in the
second field and select :guilabel:`is set`.

Click the :guilabel:`➕ (plus)` icon to the right of the rule to duplicate it.

Click the second field of the new rule, and select :guilabel:`is between` from the drop-down menu.

> Click in the date fields and use the calendar drop-down to add a start and end date. This is usually

Reviewer: ```suggestion
Click in the date fields, and use the calendar drop-down to add a start and end date. This is usually
```

the beginning and ending of the current month.

Click :guilabel:`New Rule`. Click in the new rule's first field, then type `Salesperson` in the

> search bar, or scroll through the list to select it. Click in the rule's second field and select

Reviewer: ```suggestion
search bar, or scroll through the list to select it. Click in the rule's second field, and select
```

> :guilabel:`is set` from the drop-down menu. This will exclude any results without an assigned

Reviewer: ```suggestion
:guilabel:`is set` from the drop-down menu. This excludes any results without an assigned
```

salesperson.

.. image:: expected_revenue_report/custom-filters.png
   :align: center
   :alt: Add Custom Filters pop-up window with custom filters configured for expected revenue
         report.

.. important::
   At the top of the :guilabel:`Add Custom Filter` form, there is an option to match :guilabel:`any`
   or :guilabel:`all` of the rules. In order to properly run the report, only records that match
   **all** of the following filters should be included. Before adding the filters, make sure
   :guilabel:`all` is selected in this field.

   .. image:: expected_revenue_report/match-all-filters.png
      :align: center
      :alt: Emphasis on the match all filters option on the Add Custom Filter pop-up window.

At the bottom of the :guilabel:`Add Custom Filter` form, click :guilabel:`Add`.

Save as a favorite
------------------

> To save this report as a favorite, click on the arrow to the right of the search bar, then click

Reviewer: ```suggestion
To save this report as a favorite, click on the :guilabel:`down arrow` icon to the right of the search bar, then click
```

:guilabel:`Save current search`. Enter a name for this report, then click :guilabel:`Save`.

.. tip::
   Before saving, additional selections can be made from this drop-down menu. For example, to view

>    expected revenue by individual salesperson, selected :guilabel:`Salesperson` under the

Reviewer: ```suggestion
   expected revenue by individual salespeople, selected :guilabel:`Salesperson` under the
```

   :guilabel:`Group By` heading.

View options
============

The expected revenue report benefits from utilizing multiple views. The default graph view can be
used to identify which salespeople are expected to bring in the most revenue, while the list view
and pivot view provide more detail on specific deals.

.. tabs::

   .. tab:: Graph view

>       The *graph view* is used to visualize data, and is beneficial in identifying patters and

Reviewer: ```suggestion
      The *graph view* is used to visualize data, and is beneficial in identifying patterns and
```

>       trends. *Bar charts* are used to show the distribution of data across several categories or
>       among several salespeople. *Line charts* are useful to show changing trends over a period of
>       time. *Pie charts* are useful to show the distribution or comparison of data among a small
>       number of categories or salespeople, specifically how they form the meaningful part of a whole
>       picture.

Reviewer: ```suggestion
      trends. 
      
      *Bar charts* are used to show the distribution of data across several categories or among
      several salespeople. *Line charts* are useful to show changing trends over a period of time.
      *Pie charts* are useful to show the distribution, or comparison, of data among a small number
      of categories or salespeople, specifically how they form the meaningful part of a whole
      picture.
```

      The default view for the expected revenue report is the bar chart, stacked. To change to a
      different graph view, click one of the icons at the top-left of the report. While both the

>       line chart bar chart are available in stacked view, the pie chart is not.

Reviewer: ```suggestion
      line chart and bar chart are available in stacked view, the pie chart is not.
```

      .. figure:: expected_revenue_report/graph-view-icons.png
         :align: center
         :alt: Close up view of the graph icons on the Pipeline analysis report in the CRM app.

>          Graph view icons in order: bar chart, line chart, pie chart, stacked.

Reviewer: do you think it would be beneficial to add GUIs or ** here? Or do you think it's good as-is? Curious to see what you think.

Author: I don't think that the guis or italics are necessary in the caption since it is just annotating the figure, not acting the same as normal text

   .. tab:: List view

      The *list view* provides a list of all leads that are expected to close by the designated
      date. Clicking on a lead in list view opens the record for detailed analysis, but many
      insights can be gleaned from the basic view.

      To switch to the list view, click the :guilabel:`≣ (bars)` icon at the top-right of the
      report.

      .. image:: expected_revenue_report/list-icon.png
         :align: center
         :alt: Close up of the list view icon in the CRM app.

      To add additional metrics to the report, click the *additional options menu* indicated by the
      :guilabel:`sliders` icon at the top-right of the list. Select any additional metrics from the
      drop-down menu to add them to the list view. Some options that may be useful are
      :guilabel:`Expected Closing` and :guilabel:`Probability`.

   .. tab:: Pivot view

      The *pivot view* arranges all leads that are expected to close by the designated date into a

>       table with salespeople along the Y axis and stages along the X axis.

Reviewer: ```suggestion
      table, with salespeople along the Y axis and stages along the X axis.
```

      To switch to the pivot view, click the :guilabel:`Pivot` icon at the top-right of the report.

      .. image:: expected_revenue_report/pivot-view-icon.png
         :align: center
         :alt: Close up of the pivot view icon in the CRM app.

      To add additional measures to the report, click the :guilabel:`Measures` button at the
      top-left of the report. Select any additional metrics from the drop-down menu.

      Click :guilabel:`Insert in Spreadsheet` to add the pivot view into an editable spreadsheet
      format within the *Dashboards* app. If the Odoo *Documents* app is installed, the report can

>       be inserted into a blank or existing spreadsheet and exported.

Reviewer: ```suggestion
      be inserted into a blank or existing spreadsheet, and exported.
```

   without an assigned salesperson.
 - :ref:`Specific sales teams <expected_revenue_report/sales-team>`: limits results to only include
   leads assigned to one or more sales teams. This filter is optional and should not be included if
   the report is intended for the entire company.

.. _expected_revenue_report/closing-date:

Add filter for expected closing date
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

On the :guilabel:`Add Custom Filter` pop-up window, click into the first field of the new rule.
Type `Expected Closing` into the :guilabel:`Search...` bar, or scroll to select it from the list.
Click in the second field and select :guilabel:`is set`. This limits the results to only include
leads where an estimated closing date is listed.

Next, click the :guilabel:`➕ (plus)` icon to the right of the rule to duplicate it.

.. tip::
   Using the :guilabel:`➕ (plus)` icon makes it easy to add multiple rules based on the same
   filter.

In the second field of the new rule, select :guilabel:`is between` from the drop-down menu. This
creates a set time frame during which the expected closing date must occur for leads to be included
in the results.

> Click in the date fields, and use the calendar drop-down to add a start and end date. This is

Reviewer 2: - this element is called a *popover* window in web design.
- added some more nuanced instruction here to make it super clear 😉 

```suggestion
Click in each date field, one at a time, and use the calendar popover window to add both a start and
end date to the rule. This is
```

usually the beginning and ending of the current month, or fiscal quarter.

.. _expected_revenue_report/unassigned-leads:

Exclude unassigned leads
~~~~~~~~~~~~~~~~~~~~~~~~

> After filtering for the closing date, add a :guilabel:`New Rule`. Then, click into the new rule's

Reviewer 2: ```suggestion
After filtering for the expected closing date, add a :guilabel:`New Rule`. Then, click into the new rule's
```

first field, and type `Salesperson` in the :guilabel:`Search...` bar, or scroll through the list to
select it. Click in the rule's second field and select :guilabel:`is set` from the drop-down menu.
This excludes any results without an assigned salesperson.

.. _expected_revenue_report/sales-team:

Add a filter for sales teams
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

.. note::
   This filter is optional. To view results for the entire company, do **not** add this filter, and
   continue to :ref:`View results <expected_revenue_report/view-results>`.

To limit the results of the report to one or more sales teams, click :guilabel:`New Rule`. Next,
click the first field for the new rule, and type `Sales Team` in the :guilabel:`Search...` bar, or
scroll to search through the list to locate it.

In the rule's second field, select :guilabel:`is in` from the drop-down menu. Selecting this

> operator limits results to the sales teams selected in the next field.

Reviewer 2: nitpick: the next field isn't *exactly* a drop-down menu, meaning, it's possible to mess this up.
```suggestion
operator limits results to the sales teams indicated in the next field.
```

> Lastly, in the third field, select the desired sales team from the drop-down menu. Multiple teams
> can be added in this field, where each parameter is treated with an "or" (e.g. "any") operator in
> the search logic.

Reviewer 2: suggestion to add more context + highlight the last part as a **tip** so it's super visible

```suggestion
Lastly, click into the third field, and either: make selection(s) from the complete list revealed in
the popover menu, or type the first few characters of the specific sales team's title to quickly
find and select it as a parameter.

.. tip::
   Multiple teams can be added to the `Sales Team` rule, where each parameter is treated with an
   "or" (e.g. "any") operator in the search logic.
```

      The *list view* provides a list of all leads that are expected to close by the designated
      date. Clicking on a lead in list view opens the record for detailed analysis, but many
      insights can be gleaned from the basic view.

      To switch to the list view, click the :guilabel:`≣ (list)` icon at the top-right of the
      report.

      .. image:: expected_revenue_report/list-icon.png
         :align: center
         :alt: Close up of the list view icon in the CRM app.

      To add additional metrics to the report, click the *additional options menu* indicated by the
      :guilabel:`toggle` icon at the top-right of the list.

      .. figure:: expected_revenue_report/toggle-icon.png
         :align: center
         :alt: Close up of the toggle icon in the CRM app.

         Clicking the toggle icon in *list view* opens the *additional options menu*.

      Select any additional metrics from the drop-down menu to add them to the list view. Some
      options that may be useful are :guilabel:`Expected Closing` and :guilabel:`Probability`.

   .. tab:: Pivot view

      The *pivot view* arranges all leads that are expected to close by the designated date into a

>       table, with salespeople along the Y axis and stages along the X axis.

Reviewer 2: > with salespeople along the Y axis

Doesn't appear to necessarily be true (see screenshot below):
 ![Odoo-Pipeline-Analysis(1)](https://github.com/odoo/documentation/assets/5496809/d78e009e-df13-43a2-862b-9eb766c08dc9)

We'll want to specify what the *default* is for the pivot table and how these axis values can be manipulated via the parameters in the popover menus.

As well, I'd argue _pivot table_ is going to be the **most important** view since it allows sales managers to format data quickly and nicely before injecting into a spreadsheet of some kind (hopefully Odoo _Spreadsheet_ 😉) for further analysis.
